### ISSUE TITLE
UnicodeEncodeError

### ISSUE THREAD

**Issue by Chartiza** [chklovski/CheckM2#28]
Hi all!

When I run the checkm2 program It seems like all the calculation finished, but the final results is not writing because of UnicodeEncodeError problem. 

Maybe someone faced the same problem and know how to fix it?

[10/27/2022 02:01:40 PM] INFO: Running quality prediction workflow with 30 threads.
[10/27/2022 02:01:44 PM] INFO: Calling genes in 109 bins with 30 threads:
    Finished processing 109 of 109 (100.00%) bins.
[10/27/2022 02:03:32 PM] INFO: Calculating metadata for 109 bins with 30 threads:
    Finished processing 109 of 109 (100.00%) bin metadata.
[10/27/2022 02:03:34 PM] INFO: Annotating input genomes with DIAMOND using 30 threads
Traceback (most recent call last):
  File "/gpfs/space/home/pantiukh/.conda/envs/checkm2/bin/checkm2", line 4, in <module>
    __import__('pkg_resources').run_script('CheckM2==0.1.3', 'checkm2')
  File "/gpfs/space/home/pantiukh/.conda/envs/checkm2/lib/python3.6/site-packages/pkg_resources/__init__.py", line 651, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/gpfs/space/home/pantiukh/.conda/envs/checkm2/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1448, in run_script
    exec(code, namespace, namespace)
  File "/gpfs/space/home/pantiukh/.conda/envs/checkm2/lib/python3.6/site-packages/CheckM2-0.1.3-py3.6.egg/EGG-INFO/scripts/checkm2", line 154, in <module>
    predictor.prediction_wf(args.genes, mode, args.dbg_cos, args.dbg_vectors, args.stdout)
  File "/gpfs/space/home/pantiukh/.conda/envs/checkm2/lib/python3.6/site-packages/CheckM2-0.1.3-py3.6.egg/checkm2/predictQuality.py", line 103, in prediction_wf
    diamond_out = diamond_search.run(prodigal_files)
  File "/gpfs/space/home/pantiukh/.conda/envs/checkm2/lib/python3.6/site-packages/CheckM2-0.1.3-py3.6.egg/checkm2/diamond.py", line 118, in run
    self.__call_diamond(protein_chunks, diamond_out)
  File "/gpfs/space/home/pantiukh/.conda/envs/checkm2/lib/python3.6/site-packages/CheckM2-0.1.3-py3.6.egg/checkm2/diamond.py", line 73, in __call_diamond
    sequenceClasses.SeqReader().write_fasta(seq_object, temp_diamond_input.name)
  File "/gpfs/space/home/pantiukh/.conda/envs/checkm2/lib/python3.6/site-packages/CheckM2-0.1.3-py3.6.egg/checkm2/sequenceClasses.py", line 104, in write_fasta
    fout.write('>' + seqId + '\n')
UnicodeEncodeError: 'ascii' codec can't encode character '\u03a9' in position 10: ordinal not in range(128)

**Comment by chklovski** [comment]
Hi, sorry for the late reply. There seems to be an issue with writing FASTA files to the disc during the diamond search - possibly there are permission issues or invalid characters in the provided FASTA file. Please feel free to reopen if this is still an issue.

**Comment by isidoramorel** [comment]
Hello!
I am unfortunately having the same problem. Even when I try running the testrun. 

```
[02/28/2023 02:34:10 PM] INFO: Test run: Running quality prediction workflow on test genomes with 1 threads.
[02/28/2023 02:34:10 PM] INFO: Running checksum on test genomes.
[02/28/2023 02:34:11 PM] INFO: Checksum successful.
[02/28/2023 02:34:13 PM] INFO: Calling genes in 3 bins with 1 threads:
    Finished processing 3 of 3 (100.00%) bins.
[02/28/2023 02:34:57 PM] INFO: Calculating metadata for 3 bins with 1 threads:
    Finished processing 3 of 3 (100.00%) bin metadata.
[02/28/2023 02:34:58 PM] INFO: Annotating input genomes with DIAMOND using 1 threads
Traceback (most recent call last):
  File "/home/imorel/miniconda3/envs/checkm2/bin/checkm2", line 265, in <module>
    predictor.prediction_wf(False, 'auto', False, False, False)
  File "/home/imorel/miniconda3/envs/checkm2/lib/python3.6/site-packages/checkm2/predictQuality.py", line 135, in prediction_wf
    diamond_out = diamond_search.run(prodigal_files)
  File "/home/imorel/miniconda3/envs/checkm2/lib/python3.6/site-packages/checkm2/diamond.py", line 119, in run
    self.__call_diamond(protein_chunks, diamond_out)
  File "/home/imorel/miniconda3/envs/checkm2/lib/python3.6/site-packages/checkm2/diamond.py", line 74, in __call_diamond
    sequenceClasses.SeqReader().write_fasta(seq_object, temp_diamond_input.name)
  File "/home/imorel/miniconda3/envs/checkm2/lib/python3.6/site-packages/checkm2/sequenceClasses.py", line 104, in write_fasta
    fout.write('>' + seqId + '\n')
UnicodeEncodeError: 'ascii' codec can't encode character '\u03a9' in position 6: ordinal not in range(128)
```

**Comment by isidoramorel** [comment]
And I finally solved it :) It was my python version (3.6)
with `conda create -n checkM2 -c bioconda -c conda-forge checkm2 python=3.7` worked perfectly.

**Comment by FWittmers** [comment]
@chklovski, I have been running into the same error. It can't be a fasta file since I error out even when I just try the testrun. Tried multiple things, including the change in python-version suggested above, but that doesnt't solve it either...any other ideas? 
(This assumes that the testrun uses fasta files that you know are clean)
